### ISSUE TITLE
ruleset.count() not changing after applying ruleset.exrule()

### ISSUE THREAD

**Issue by anandkr07** [dateutil/dateutil#104]
Seems like the ruleset.count() is getting cached after first call to it. An Example of this behavior -

    >>> rules = rrule.rruleset()
    >>> rules.rrule(rrule.rrule(rrule.DAILY, until=datetime.datetime(2038,1,1,0,0,0), dtstart=datetime.datetime(now.year,now.month,now.day,0,0,0)))
    >>> rules.count()
    8179
    >>> rules.exrule(rrule.rrule(rrule.DAILY, until=datetime.datetime(2038,1,1,0,0,0), dtstart=datetime.datetime(now.year,now.month,now.day,0,0,0)))
    >>> rules.count()
    8179

Only after iterating over the ruleset , do we get the correct value , Example -

    >>> list(rules)
    []
    rules.count()
    0

**Comment by pganssle** [comment]
The issue here is that `rrulebase` caches the length after the first call to `count()`, and this cache is not invalidated when changes are made to the rruleset:

```python
from datetime import datetime as dt
from dateutil import rrule as rr

dts = dt(2015, 8, 11)
dte = dt(2016, 8, 11)

crr = rr.rrule(rr.DAILY, dtstart=dts, until=dte)

rules = rr.rruleset()
print(rules.count())
>>> 0

rules.rrule(crr)
print(rules.count())
>>> 0

rules._len = None
print(rules.count())
>>> 367
```
This should be easy enough to fix, I can make sure it's fixed in 2.5.0. I am in general loathe to recommend building anything upon private methods or attributes, but in the meantime, anything that iterates over the rruleset should update the count, so `list(rules)` or something like: ```for _ in rules: pass``` (this is what `rruleset.count()` would be doing anyway if the cache were properly invalidated anyway, so it shouldn't degrade performance).

**Comment by pganssle** [comment]
Pushing fix to 2.5.1 to speed release of 2.5.0.
